event/src/unvalidated/builder.rs

```diff
@@ -319,15 +468,42 @@ mod tests {
 
         let envelope_cbor_str = multibase::encode(
             multibase::Base::Base64Url,
-            &signed_event.encode_envelope().unwrap(),
+            signed_event.encode_envelope().unwrap(),
         );
 
         assert_eq!(SIGNED_INIT_EVENT, envelope_cbor_str);
 
         let event_car_str = multibase::encode(
             multibase::Base::Base64Url,
-            &signed_event.encode_car().await.unwrap(),
+            signed_event.encode_car().await.unwrap(),
         );
         assert_eq!(SIGNED_INIT_EVENT_CAR, event_car_str);
     }
+    #[tokio::test]
+    async fn build_time_event() {
+        let id = Cid::from_str(SIGNED_INIT_EVENT_CID).unwrap();
+        let prev =
+            Cid::from_str("bagcqcerae5oqoglzjjgz53enwsttl7mqglp5eoh2llzbbvfktmzxleeiffbq").unwrap();
+        let tx_hash =
+            Cid::from_str("bagjqcgzadp7fstu7fz5tfi474ugsjqx5h6yvevn54w5m4akayhegdsonwciq").unwrap();
+
+        let metadata_cid =
+            Cid::from_str("bafyreifjkogkhyqvr2gtymsndsfg3wpr7fg4q5r3opmdxoddfj4s2dyuoa").unwrap();
+        let event = Builder::time()
+            .with_id(id)
+            .with_tx(
+                "eip155:11155111".to_string(),
+                tx_hash,
+                "f(bytes32)".to_string(),
+            )
+            .with_root(0, ipld! {[prev, Ipld::Null, metadata_cid]})
+            .build()
+            .unwrap();
+
+        let event_car_str = multibase::encode(
+            multibase::Base::Base64Url,
+            event.encode_car().await.unwrap(),
+        );
+        assert_eq!(TIME_EVENT_CAR, event_car_str);
+    }
```

Comment:
it would be nice to have a unit test showing how this works with a tree that has intermediate nodes not just the root.  But low priority and not worth delaying the rest of the stack of PRs this is depending on, so feel free to just add a TODO for now.

Reply:
FWIW the migration tests do exercise the code paths for 10 deep trees.